lib/Conversion/TorchToMhlo/Basic.cpp

```diff
@@ -769,6 +769,38 @@ LogicalResult ConvertAtenOp<AtenContiguousOp>::matchAndRewrite(
   return success();
 }
 
+// AtenCopyOp
+template <>
+LogicalResult ConvertAtenOp<AtenCopyOp>::matchAndRewrite(
+    AtenCopyOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  Value self = adaptor.getSelf();
+  Value src = adaptor.getSrc();
+  auto selfType = self.getType().cast<RankedTensorType>();
+  auto srcType = src.getType().cast<RankedTensorType>();
+
+  if (!selfType.hasStaticShape() || !srcType.hasStaticShape()) {
+    return op.emitError(
+        "unimplemented: only tensor types with static shape are supported");
+  }
+
+  // The non_blocking should be a constant `False`.
+  bool nonBlocking;
+  if (!matchPattern(op.getNonBlocking(), m_TorchConstantBool(&nonBlocking))) {
+    return rewriter.notifyMatchFailure(
+        op, "unimplemented: non_blocking must be a constant");
+  } else if (nonBlocking) {
+    return rewriter.notifyMatchFailure(
+        op, "unimplemented: non_blocking is expected to be false");
+  }
```

Comment:
Nit: there is no need to have a check here. The memory layout and semantics can be discarded.

Reply:
Okay, there is no check in the new version.